server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/put/ClusterPutWeightedRoutingRequest.java

```diff
@@ -0,0 +1,165 @@
+                NamedXContentRegistry.EMPTY,
+                DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+                source,
+                contentType
+            )
+        ) {
+            String attrValue = null;
+            Map<String, Double> weights = new HashMap<>();
+            Double attrWeight = null;
+            XContentParser.Token token;
+            // move to the first alias
+            parser.nextToken();
+            while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                if (token == XContentParser.Token.FIELD_NAME) {
+                    attrValue = parser.currentName();
+                } else if (token == XContentParser.Token.VALUE_STRING) {
+                    attrWeight = Double.parseDouble(parser.text());
+                    weights.put(attrValue, attrWeight);
+                } else {
+                    throw new OpenSearchParseException(
+                        "failed to parse weighted routing request attribute [{}], " + "unknown type",
+                        attrWeight
+                    );
+                }
+            }
+            this.weightedRouting = new WeightedRouting(this.attributeName, weights);
+        } catch (IOException e) {
+            logger.error("error while parsing put for weighted routing request object", e);
+        }
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (weightedRouting == null) {
+            validationException = addValidationError("Weighted routing request object is null", validationException);
+        }
+        if (weightedRouting.attributeName() == null || weightedRouting.attributeName().isEmpty()) {
+            validationException = addValidationError("Attribute name is missing", validationException);
+        }
+        if (weightedRouting.weights() == null || weightedRouting.weights().isEmpty()) {
+            validationException = addValidationError("Weights are missing", validationException);
+        }
+        int countValueWithZeroWeights = 0;
+        double weight;
+        try {
+            for (Object value : weightedRouting.weights().values()) {
+                if (value == null) {
+                    validationException = addValidationError(("Weight is null"), validationException);
+                } else {
+                    weight = Double.parseDouble(value.toString());
+                    countValueWithZeroWeights = (weight == 0) ? countValueWithZeroWeights + 1 : countValueWithZeroWeights;
+                }
+            }
+        } catch (NumberFormatException e) {
+            validationException = addValidationError(("Weight is not a number"), validationException);
+        }
+        if (countValueWithZeroWeights > 1) {
+            validationException = addValidationError(("More than one value has weight set as 0 "), validationException);
```

Comment:
lets also add the count explicitly

Reply:
Do you mean this value should be read from dynamic settings?